client/auth_rep.go

```diff
@@ -61,29 +58,26 @@ func (client *ThreeScaleClient) authRep(values url.Values)(ApiResponse, error) {
 	return resp, nil
 }
 
-
-
-// Create valid params for AuthRep
-func NewAuthRepParams(key string, referrer string, userId string) AuthRepParams {
+func NewAuthRepParamsAppID(key string, referrer string, userId string, metrics Metrics, log Log) AuthRepParams {
 	return AuthRepParams{
 		AuthorizeParams: AuthorizeParams{
 			AppKey:   key,
 			Referrer: referrer,
 			UserId:   userId,
-			Metrics:  make(Metrics),
+			Metrics:  metrics,
 		},
-		Log: make(Log),
+		Log: log,
 	}
 }
 
-// Create valid params for AuthRepKey
-func NewAuthRepKeyParams(referrer string, userId string) AuthRepKeyParams {
-	return AuthRepKeyParams{
-		AuthorizeKeyParams: AuthorizeKeyParams{
+func NewAuthRepParamsUserKey(referrer string, userId string, metrics Metrics, log Log) AuthRepParams {
+	return AuthRepParams{
```

Comment:
@jmprusi curios why we changed the function signature here? Whats the benefit? If it makes sense then its not a problem but previously the user was abstracted from having to make the internal types

Reply:
I really prefer to make that explicit, instead of initializing the metrics and log to empty, allow the user to decide.

This feels better to me, but we can change it.

Reply:
👍  mot a deal breaker for me regardless, was just wondering. Lets see what @unleashed thinks. Other than that changes look fine.

Reply:
As we talked, better to go with explicit parameters, so let's keep it this way.

What's missing from the PR so we can merge? @unleashed @PhilipGough

Reply:
Re: merging, it's ok to me. What I find missing is actually removing this `Log` parameter, not even if the callers use `nil`, such that any caller who _wanted_ to use `Log` would then add it to the data in `AuthRepParams`. I wish I could make everyone forget there is a `log` parameter to any 3scale call.

Reply:
We can remove it on another PR. and just deprecate it client side :)